certbot/account.py

```diff
@@ -175,33 +175,44 @@ def _find_all_for_server_path(self, server_path):
             except errors.AccountStorageError:
                 logger.debug("Account loading problem", exc_info=True)
 
-
         if not accounts and server_path in constants.LE_REUSE_SERVERS:
             # find all for the next link down
             prev_server_path = constants.LE_REUSE_SERVERS[server_path]
             prev_accounts = self._find_all_for_server_path(prev_server_path)
             # if we found something, link to that
             if prev_accounts:
-                if os.path.islink(accounts_dir):
-                    os.unlink(accounts_dir)
-                else:
-                    try:
-                        os.rmdir(accounts_dir)
-                    except OSError:
-                        return []
-                prev_account_dir = self.config.accounts_dir_for_server_path(prev_server_path)
-                os.symlink(prev_account_dir, accounts_dir)
+                try:
+                    self._symlink_to_accounts_dir(prev_server_path, server_path)
+                except OSError:
+                    return []
             accounts = prev_accounts
         return accounts
 
     def find_all(self):
         return self._find_all_for_server_path(self.config.server_path)
 
+    def _symlink_to_accounts_dir(self, prev_server_path, server_path):
+        server_path = self.config.server_path
```

Comment:
This line immediately overwrites the `server_path` parameter passed to this function. I think we should just delete this line as it should allow these methods to just work if there's ever something like an ACMEv3 server that's backwards compatible with both v2 and v1 accounts.

Reply:
Nice catch, this is a mistake caused by too much moving code around. The entire point of this function is to pass in the `server_path` parameter...